core/8_bubble_sort_algo/src/main.rs

```diff
@@ -0,0 +1,17 @@
+fn bubble_sort(arr :&mut [i32]){
+    let n = arr.len();
+    for i in 0..n{
+        for j in 0..n-i-1{
+            if arr[j]>arr[j+1]{
+                arr.swap(j,j+1);
```

Comment:
Here we detect a `built in function`, never know rust array have swap, wondering if we can swap two values by their reference manually rather than using than the `swap` method

Reply:
I tried implementing my on swap function like this:

```rust
fn bubble_sort(arr :&mut [i32]){
    let n = arr.len();
    for i in 0..n{
        for j in 0..n-i-1{
            if arr[j]>arr[j+1]{
               // ERROR:  cannot borrow `arr[_]` as mutable more than once at a time
                swap(&mut arr[j], &mut arr[j+1]);
            }
        }
    }
}

fn swap(a: &mut i32, b: &mut i32){
    let c =  std::mem::take(a);
    *a = std::mem::take(b);
    *b = c;
}

```
Have to figure it out later, good for now tho!